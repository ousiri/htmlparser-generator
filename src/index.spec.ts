
import { parseDOM } from 'htmlparser2'
import { Element } from "domhandler";
import generator from '.'

// Add an `attributes` prop to the Element for now, to make it possible for Jest to render DOM nodes.
Object.defineProperty(Element.prototype, "attributes", {
  get() {
    return Object.keys(this.attribs).map(name => ({
      name,
      value: this.attribs[name]
    }));
  },
  configurable: true,
  enumerable: false
});

describe("Generator", () => {
  test("parseDOM", () => {
    const dom = parseDOM("<a foo><b><c><?foo>Yay!")
    generator(dom)
    expect(dom).toMatchSnapshot()
  })
})