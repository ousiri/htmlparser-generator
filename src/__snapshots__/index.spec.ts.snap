// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generator parseDOM 1`] = `
Array [
  <a
    foo=""
  >
    <b>
      <c>
        ProcessingInstruction {
          "data": "?foo",
          "endIndex": null,
          "name": "?foo",
          "next": DataNode {
            "data": "Yay!",
            "endIndex": null,
            "next": null,
            "parent": <c>
              [Circular]
              [Circular]
            </c>,
            "prev": [Circular],
            "startIndex": null,
            "type": "text",
          },
          "parent": <c>
            [Circular]
            DataNode {
              "data": "Yay!",
              "endIndex": null,
              "next": null,
              "parent": <c>
                [Circular]
                [Circular]
              </c>,
              "prev": [Circular],
              "startIndex": null,
              "type": "text",
            }
          </c>,
          "prev": null,
          "startIndex": null,
          "type": "directive",
        }
        DataNode {
          "data": "Yay!",
          "endIndex": null,
          "next": null,
          "parent": <c>
            ProcessingInstruction {
              "data": "?foo",
              "endIndex": null,
              "name": "?foo",
              "next": [Circular],
              "parent": <c>
                [Circular]
                [Circular]
              </c>,
              "prev": null,
              "startIndex": null,
              "type": "directive",
            }
            [Circular]
          </c>,
          "prev": ProcessingInstruction {
            "data": "?foo",
            "endIndex": null,
            "name": "?foo",
            "next": [Circular],
            "parent": <c>
              [Circular]
              [Circular]
            </c>,
            "prev": null,
            "startIndex": null,
            "type": "directive",
          },
          "startIndex": null,
          "type": "text",
        }
      </c>
    </b>
  </a>,
]
`;
